### PR TITLE
Port Sierra RID support for 1.0.* hostpolicy

### LINKED ISSUE #538: Port Sierra RID fix to 1.0.3
From PR https://github.com/dotnet/core-setup/pull/536

CC @leecow

### PR DESCRIPTION
**Pull request by schellap**
Fixes #538

### CODE REVIEW

**Comment by schellap** [comment]
@gkhanna79 PTAL.

File: build_projects/dotnet-host-build/CompileTargets.cs [hunk 1]
```diff
@@ -264,9 +264,9 @@ public static BuildTargetResult GenerateMSbuildPropsFile(BuildTargetContext c)
             msbuildProps.AppendLine("  <PropertyGroup>");
             msbuildProps.AppendLine($"    <Platform>{arch}</Platform>");
             msbuildProps.AppendLine($"    <DotNetHostBinDir>{Dirs.CorehostLatest}</DotNetHostBinDir>");
-            msbuildProps.AppendLine($"    <HostVersion>{hostVersion.LatestHostPolicyVersion.WithoutSuffix}</HostVersion>");
+            msbuildProps.AppendLine($"    <HostVersion>{hostVersion.LatestHostVersion.WithoutSuffix}</HostVersion>");
```
Comment (gkhanna79) [inline build_projects/dotnet-host-build/CompileTargets.cs]: For consistency, we should make this change in master too.

File: build_projects/dotnet-host-build/CompileTargets.cs [hunk 2]
```diff
@@ -264,9 +264,9 @@ public static BuildTargetResult GenerateMSbuildPropsFile(BuildTargetContext c)
...
             msbuildProps.AppendLine($"    <HostResolverVersion>{hostVersion.LatestHostFxrVersion.WithoutSuffix}</HostResolverVersion>");
-            msbuildProps.AppendLine($"    <HostPolicyVersion>{hostVersion.LatestHostVersion.WithoutSuffix}</HostPolicyVersion>");
+            msbuildProps.AppendLine($"    <HostPolicyVersion>{hostVersion.LatestHostPolicyVersion.WithoutSuffix}</HostPolicyVersion>");
```
Comment (gkhanna79) [inline build_projects/dotnet-host-build/CompileTargets.cs]: Same here.

File: build_projects/shared-build-targets-utils/Utils/HostVersion.cs [hunk 3]
```diff
@@ -67,7 +67,7 @@ public override string ToString()
         public string LatestHostBuildMinor => "00";
         public VerInfo LatestHostVersion => new VerInfo(1, 0, 1, LatestHostPrerelease, LatestHostBuildMajor, LatestHostBuildMinor, CommitCountString);
         public VerInfo LatestHostFxrVersion => new VerInfo(1, 0, 1, LatestHostPrerelease, LatestHostBuildMajor, LatestHostBuildMinor, CommitCountString);
-        public VerInfo LatestHostPolicyVersion => new VerInfo(1, 0, 1, LatestHostPrerelease, LatestHostBuildMajor, LatestHostBuildMinor, CommitCountString);
+        public VerInfo LatestHostPolicyVersion => new VerInfo(1, 0, 2, LatestHostPrerelease, LatestHostBuildMajor, LatestHostBuildMinor, CommitCountString);
```
Comment (gkhanna79) [inline build_projects/shared-build-targets-utils/Utils/HostVersion.cs]: For consistency, I would suggest making this 1.0.3.

File: build_projects/dotnet-host-build/CompileTargets.cs (hunk 1 above)
Comment (gkhanna79) [inline build_projects/dotnet-host-build/CompileTargets.cs]: BTW, shouldnt this be LockedHostVersion?

Comment (schellap) [inline build_projects/dotnet-host-build/CompileTargets.cs]: We won't produce the packages. The locking is for people who consume the packages not producing the packages.

File: pkg/projects/Microsoft.NETCore.DotNetHost/win/Microsoft.NETCore.DotNetHostResolver.pkgproj [hunk 4]
```diff
@@ -13,7 +13,7 @@
   
   <ItemGroup>
     <Dependency Include="Microsoft.NETCore.DotNetHost">
-      <Version>$(HostFullVersion)</Version>
```
Comment (schellap) [inline pkg/projects/Microsoft.NETCore.DotNetHost/win/Microsoft.NETCore.DotNetHostResolver.pkgproj]: This is just renaming. No physical change -- it is confusing to have the name as `FullVersion` and have a targeted fix under the name.

Comment (gkhanna79) [inline pkg/projects/Microsoft.NETCore.DotNetHost/win/Microsoft.NETCore.DotNetHostResolver.pkgproj]: Why do we need this rename in servicing branch? I dont believe it is imperative for the fix. Please feel free to make it in master.

**Comment by schellap** [comment]
@gkhanna79 made the fix super narrow.

File: pkg/stable.packages.props [hunk 5]
```diff
@@ -17,9 +17,11 @@
     <StablePackage Include="Microsoft.NETCore.DotNetHost">
       <Version>1.0.1</Version>
     </StablePackage>
+    <!--
     <StablePackage Include="Microsoft.NETCore.DotNetHostPolicy">
-      <Version>1.0.1</Version>
+      <Version>1.0.3</Version>
```
Comment (gkhanna79) [inline pkg/stable.packages.props]: This is not required right now but only when we perform stabilization.

**Comment by gkhanna79** [comment]
Modulo a comment, LGTM.

File: pkg/stable.packages.props (hunk 5 above)
Comment (schellap) [inline pkg/stable.packages.props]: Changed, so we don't forget

**Comment by gkhanna79** [comment]
LGTM. Please send email to shiproom and merge upon approval.

**Pull request merged**